### ISSUE TITLE
Make auto-indent and auto-closing-character-insertion optional

### ISSUE THREAD

**Issue by FArekkusu** [codewars/codewars.com#2198]
After solving and translating some katas for the last few dates, I find these 2 features more annoying than helpful. It'd be great if they were made optional settings, toggleable in the account settings for example, instead of being always on.

**Comment by kazk** [comment]
Yeah, I've been wanting to make the editor configurable in general. It's impossible to make everyone happy with a single configuration.

If these new features are annoying a lot of users, I might prioritize it.

---

(Notes for future reference.)

Instead of putting it in Account Settings, I want to remove the awkward `VIM/EMACS` buttons and add :gear: that opens some dialog to set editor configurations (key maps, indent size, feature toggles, etc.).

For persisting configurations, maybe use something similar to [VS Code](https://code.visualstudio.com/docs/getstarted/settings#_default-settings) (not a subset because many values won't make sense for CodeMirror).
```json
{
  "editor.tabSize": 2,
  "editor.insertSpaces": true,
  "editor.autoClosingBrackets": true,
  "editor.autoIndent": true,
  "[rust]": {
    "editor.tabSize": 4
  },
  "[go]": {
    "editor.insertSpaces": false
  }
}
```

But this will require significantly more effort and will probably need to wait until I start working on rewriting the UI. So I might end up adding some options in Account Settings temporarily.

**Comment by kazk** [comment]
You can run the following in the developer console for now.

```javascript
localStorage.setItem('editorOptions', JSON.stringify({
  autoCloseBrackets: false,
  smartIndent: false
}));
```

The following options are supported:

- `tabSize`
- `indentWithTabs`: defaults to `false` (indent with spaces, and Tab expands to `tabSize` spaces)
- `smartIndent`
- `autoCloseBrackets`

You can also override per language like in VSCode.

```javascript
localStorage.setItem("editorOptions", JSON.stringify({
  tabSize: 2,
  indentWithTabs: false,
  "[go]": {
    tabSize: 4,
    indentWithTabs: true,
  }
}));
```

**Comment by JohanWiltink** [comment]
I don't know what's feasible and what is not, but I wholeheartedly support _not_ fiddling with what I type.

Auto-indent works for me in JS, though not in LC ( we may have done something wrong with defining the syntax ), but I positively _hate_ the auto inserting closing parens, and eating closing parens I _actually type._

Language-specific options would be nice - not all languages are the same after all.

**Comment by janAkali** [comment]
Any progress on this? I hate this stupid anti-feature - it makes me do Codewars less than I'd like.
Just an example from today:
```nim
proc dist*(v, mu: float64): float64 =
  v * (5/18) * t + (v*(5/18)^2 / (254.016 * mu)
                           ^ my cursor is to the left of closing paren
```

If I type a closing parenthesis - it doesn't work. Nah just skips over. Who in their mind wants this behavior?
Is this a bug? Should I open a new issue?

P.S. developer console trick works, but I regularly clean browser cache and always forget to redo the config.

**Comment by Madjosz** [comment]
FYI: [Polyglot user script since 1.19.0](https://github.com/hobovsky/polyglot) now provides settings options to configure the default behaviour.
